mpeg2/src/ts.rs

```diff
@@ -36,21 +39,66 @@ impl AdaptationField {
             } else {
                 None
             };
+            let mut n = if af.program_clock_reference_27mhz.is_some() { 8 } else { 2 };
```

Comment:
We also need to skip additional bytes if the OPCR is present:

```suggestion
            if buf[1] & 0x08 != 0 {
                        n += 6; // skip over the OPCR
            }            
```

Reply:
okay. I think we also need to skip `splice_countdown` if its flag is 1